packages/dashboards/src/lib/configurator/components/widgets/table/columns-editor-v2/column-configuration/presentation-configuration/presentation-configuration-v2.component.ts

```diff
@@ -0,0 +1,333 @@
+            .subscribe(updateParentForm);
+    }
+
+    public onValueChange() {
+        if (!this.changeFn) {
+            return;
+        }
+
+        this.changeFn(this.form.value);
+    }
+
+    private updateSubtitle() {
+        this.subtitleText = `${this.getSelectedFormatterDefinition()?.label}`;
+        if (this.getSelectedDataField()) {
+            this.subtitleText = this.subtitleText.concat(`, ${this.getSelectedDataField()?.label}`);
+        }
+    }
+
+    /**
+     * Creates portal for dynamic configuration of formatter value.
+     * @returns ComponentPortal
+     */
+    private createFormatterConfigurator() {
+        const formatterDefinition = this._providedFormatters.find(
+            formatter => formatter.componentType === this.form.get("componentType")?.value
+        );
+
+        // if configurationComponent property is present in formatters configuration, use it to render portal,
+        // otherwise, use default ValueSelectorComponent
+        if (formatterDefinition) {
+            this.formatterConfigurator = formatterDefinition.configurationComponent ?? "ValueSelectorComponent";
+        } else {
+            this.formatterConfigurator = null;
+        }
+
+        // if the currently selected component doesn't match the original value from the input then we reset the form values
+        if (formatterDefinition?.componentType !== this.formatter?.componentType && this.formatter?.properties?.dataFieldIds) {
+            this.formatter.properties.dataFieldIds = { value: null };
+        }
+
+        this.formatterConfiguratorProps = {
+            dataFields: this.dataFields,
+            formatter: this.formatter,
+            // TODO: Ensure that formatterDefinition is not undefined
+            formatterDefinition: formatterDefinition as IFormatterDefinition,
+        };
+    }
+
+    private updateAvailableFormatters() {
+        if (!this._dataFields?.length || !this._providedFormatters.length) {
+            return;
+        }
+
+        // allow by default RawFormatter which has null as dataType
+        const sourceDataTypes: Record<string, boolean> = { null: true };
+        this._dataFields.forEach(f => sourceDataTypes[f.dataType] = true);
+        this._formatters = this._providedFormatters.filter(
+            (f) => {
+                // cast to array in case we have a single value
+                const formatterDataTypes = f.dataTypes.value instanceof Array
+                    ? f.dataTypes.value
+                    : [f.dataTypes.value];
+
+                return formatterDataTypes.some(v => sourceDataTypes[v]);
+            }
+        );
+        if (this.form) {
+            this.createFormatterConfigurator();
+        }
+    }
+
+    private subscribeToFormattersRegistry(): void {
+        this.handleFormattersUpdate(this.formattersRegistry.getItems());
+
+        this.formattersRegistry.stateChanged$.pipe(
+            tap(this.handleFormattersUpdate.bind(this)),
```

Comment:
is there a reason this is being done in a tap instead of in the subscribe block at line 314?

Reply:
This is from v1 and it is a code style if anything.